server/src/test/java/org/opensearch/extensions/rest/ExtensionRestRequestTests.java

```diff
@@ -49,6 +51,7 @@ public void setUp() throws Exception {
         expectedMethod = Method.GET;
         expectedPath = "/test/uri";
         expectedParams = Map.ofEntries(entry("foo", "bar"), entry("baz", "42"));
+        expectedHeaders = Map.ofEntries(entry("foo", Arrays.asList("hello", "world")));
```

Comment:
I don't see tests exercising scenarios around the deny list or allow list changes you've made.  Lets make sure to add a test for this.

From a security perspective - I would love to see an integration test that is certain both Authorization headers are not included.  But at minimum unit tests that confirm the behavior.

Reply:
You would make a request against the local cluster and then when the extension retrieves the call you would verify the expected headers are there and the other values are not present.

@dbwiddis What do you think about this test?  If you are in alignment with an end to end test, could you provide guidance on how this has been done in the past or what recommendations you'd have for this test extensions functionality?

Reply:
This is overly complicated.   We just really need to test the logic of filtering the headers and extracting to a new method will do that.

And we can do an IT specifically for the authorization headers when we get IT tests up on SDK.